src/main/java/com/doit/api/user/UserService.java

```diff
@@ -10,56 +15,66 @@
 
 
 @Service
-public class UserService {
+@AllArgsConstructor
+public class UserService implements UserDetailsService {
 
     @Autowired
-    private PasswordEncoder passwordEncoder;
-
+    private final BCryptPasswordEncoder bCryptPasswordEncoder;
     @Autowired
-    private UserRepository userRepository;
+    private final UserRepository userRepository;
+
+    public final static String USER_NOT_FOUND_MSG = "user with email %s not found";
+
 
-    // TODO remove this
     @PostConstruct
     // add some users if db is empty
     void init() {
-        findByEmail("user@abc.com")
-        .map(user -> user)
-        .orElseGet(() -> createUser(
-                User.builder()
-                    .email("user@abc.com")
-                    .isEnabled(true)
-                    .isAccountNonExpired(true)
-                    .isAccountNonLocked(true)
-                    .isCredentialsNonExpired(true)
-                    .password("user")
-                    .build()));
-        findByEmail("admin@abc.com")
-        .map(user -> user)
-        .orElseGet(() -> createUser(User.builder()
-                .email("admin@abc.com")
-                .isEnabled(true)
-                .isAccountNonExpired(true)
-                .isAccountNonLocked(true)
-                .isCredentialsNonExpired(true)
-                .password("admin")
-                .build()));
-    }
 
-    // this is just an alias for spring security
-    public Optional<User> findByUsername(String username) {
-        return findByEmail(username);
-    }
+        createUser(
+                new User(
+                        "sofiene",
+                        "thabet",
+                        "sof@gmail.com",
+                        "1234",
+                        UserRole.USER
+                )
+        );
```

Comment:
We should add new users only when they don't exist. Hence the use of `find(...).map(...).orElse(...)`. Otherwise exceptions are thrown each time.

Reply:
Not a problem.
Exceptions are already handled by createUser() method.

Reply:
Actually `createUser()` throws an `IllegalStateException`, it does not handle it.

Reply:
Indeed.But I changed it to a String msg.
 if (userExists) {
            return("email already exists.");
        }